arch/posix/core/posix_core.c

```diff
@@ -507,16 +507,11 @@ void z_impl_k_thread_abort(k_tid_t thread)
 			thread_idx,
 			__func__);
 
-		if (arch_is_in_isr()) {
-			z_thread_single_abort(thread);
-			return;
-		}
-
-		z_self_abort();
-		CODE_UNREACHABLE; /* LCOV_EXCL_LINE */
+		z_thread_abort(_current);
+		return;
```

Comment:
Actually yes, in interrupt context.  This is the native_posix implementation of k_thread_abort() and it will be called in all the same contexts the "official" one is.  But you're right, the ISR test is already covered above so it's silly to have the return here.